plugins/flat-object/src/main/java/org/opensearch/flatobject/xcontent/KeyValueJsonXContentParser.java

```diff
@@ -0,0 +1,249 @@
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentLocation;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.common.xcontent.support.AbstractXContentParser;
+import org.opensearch.index.mapper.ParseContext;
+
+import java.io.IOException;
+import java.nio.CharBuffer;
+import java.util.logging.Logger;
+
+public class KeyValueJsonXContentParser extends AbstractXContentParser {
+    private XContentParser parser;
+    private XContentBuilder builder = XContentBuilder.builder(JsonXContent.jsonXContent);
+    private ParseContext parseContext;
+
+    private NamedXContentRegistry xContentRegistry;
+
+    private DeprecationHandler deprecationHandler;
+    /**
+     * logging function
+     */
+
+    private static final Logger logger = Logger.getLogger((KeyValueJsonXContentParser.class.getName()));
+
+    public KeyValueJsonXContentParser(
+        NamedXContentRegistry xContentRegistry,
+        DeprecationHandler deprecationHandler,
+        ParseContext parseContext
+    ) throws IOException {
+        super(xContentRegistry, deprecationHandler);
+        this.parseContext = parseContext;
+        this.deprecationHandler = deprecationHandler;
+        this.xContentRegistry = xContentRegistry;
+        this.parser = parseContext.parser();
+    }
+
+    public XContentParser parseObject() throws IOException {
+        builder.startObject();
+        parseToken();
+        builder.endObject();
+        String jString = XContentHelper.convertToJson(BytesReference.bytes(builder), false, XContentType.JSON);
+        logger.info("Before createParser, jString: " + jString + "\n");
+
+        return JsonXContent.jsonXContent.createParser(this.xContentRegistry, this.deprecationHandler, String.valueOf(jString));
+    }
+
+    private void parseToken() throws IOException {
+        String currentFieldName;
+        while (this.parser.nextToken() != Token.END_OBJECT) {
+
+            currentFieldName = this.parser.currentName();
+
+            logger.info("currentFieldName: " + currentFieldName + "\n");
+            StringBuilder parsedFields = new StringBuilder();
+            if (this.parser.nextToken() == Token.START_OBJECT){
+                /**
+                 * for nested Json, make a copy of parser, then parse the entire Json as string.
+                 * for example:
+                 * {"grandpa": {
+                 *     "dad": {
+                 *     "son": "me"
+                 * } }
+                 * the flat field for "grandpa" would be {"grandpa_path": "grandpa", "grandpa_value"= "{"dad: {"son": "me"}"}"}
+                 */
+                //To do. to convert the entire JsonObject without changing the tokenizer position.
```

Comment:
Is that what you want? 

I thought there should be something like:
`"grandpa_path":"grandpa.dad.son=me"` and `"grandpa_value":"me"`, so you could search by the fully-qualified path or by values alone, depending on whether or not the query has the path qualifier.

Reply:
This is what the code does now: 
"grandpa_path":"grandpa",
"grandpa_value":"{"dad: {"son": "me"}"}".
"dad_path": "grandpa.dad"
"dad_value":  "{"son": "me"}"
"son_path": "grandpa.dad.son"
"son_value": "me"
 
 And yes, this idea currently doesn't make the doc path works in query yet. I am still thinking through it

Reply:
I made the new commit  [parse path_content as string field](https://github.com/mingshl/OpenSearch-Mingshl/pull/3/commits/0d09e37eff7219a7829b1eaec6a38410da4a471c) to parse the pathAndValue. 

so it parses like
                 /*
                 * grandpa: {
                 * "grandpa_key": "grandpa",
                 * "grandpa_value"= "{dad: {son: me}}} ,
                 * "grandpa_pathAndValue =  "grandpa={"dad: {son: me}}}"
                 * "dad_key" = "dad",
                 * "dad_value"= "{son: me}",
                 * "dad_pathAndValue =  "grandpa.dad={son: me}}"
                 * "son_key" = "son",
                 * "son_value"= "me",
                 * "son_pathAndValue =  "grandpa.dad.son=me" }"
                 *
                 */

But I realized that I only create one field, and query with in string field might takes longer time. and I better split up the 'key', 'value' in one string field, and 'pathAndValue' in another stringfield, so in dotpath query, it at least do bisection method and only need to scan half of the term to get result back. Thinking of a good way to create two string fields..